### DISCUSSION TITLE
LogBatch::startBatch(), keep LogBatch openend during jobs

### DISCUSSION THREAD

**Discussion by centiveo** [spatie/laravel-activitylog#915] [General]
I use a Bus::batch to go through some queue jobs. I want to log all the events during the different jobs in my bus into the same LogBatch. Now, the LogBatch is not opened within a task. It would be nice if I can use the LogBatch to log all actions during the Bus::batch-process. Or is that already possible?

**Comment by Gummibeer** [comment]
Hey,
Right now it would require some customization and hacking but possible.
I think that a PR removing the hacking should be fine.

Idea:

Open an activity batch before you start the queue batch or even easier - generate a new UUID `Str::uuid()`.
Take that UUID and pass it to every batch job as Part of the payload so you have access to `$this->activityBatchUuid` (name it how you want) in each job.
At the start of each job you check if the property is filled and if so, now the hacking, set the provided UUID in the `LogBatch` class as the UUID and likely increment the batch counter as well.
This could be done by extending the package class and rebind it as Singleton to the container.

I think that a method like `setBatch(string $uuid)` would be fine to PR. @nagi1 ?
This method should set the UUID and always set the counter to `1`.

@awebartisan had the need as well to have a general queue batch payload shared across all jobs. I'm not sure if he has found any solution? 🤔

**Reply by nagi1** [reply]
It would make sense, I'm happy that someone found a really good usecase for batches!

Working on it @Gummibeer :fire:

**Comment by nagi1** [comment]
#918
